loadtest/loadtest.go

```diff
@@ -75,26 +77,42 @@ func (lt *LoadTester) AddUsers(numUsers int) (int, error) {
 // addUser is an internal API called from Run and AddUsers both.
 // DO NOT call this by itself, because this method is not protected by a mutex.
 func (lt *LoadTester) addUser() error {
-	activeUsers := len(lt.controllers)
+	activeUsers := len(lt.activeControllers)
 	if activeUsers == lt.config.UsersConfiguration.MaxActiveUsers {
 		return ErrMaxUsersReached
 	}
-	userId := activeUsers + 1
+
+	idleUsers := len(lt.idleControllers)
+	userId := activeUsers + idleUsers + 1
```

Comment:
This is a bit confusing cause wouldn't `idleUsers` always be 0 if we end up using this value?
Also I'd prefer to see the assignment in the block where it's needed.

Reply:
Well, it is used 2 times, therefore I didn't mind it. I didn't get why `idleUsers` will be `0` always?

Reply:
That value is only used in the `if controller == nil` block. If we get there it means there are no idle controllers. That was my reasoning.

Reply:
I see. nice catch!